src/devices/Ft232H/Ft232HDevice.cs

```diff
@@ -0,0 +1,857 @@
+            {
+                _gpioLowDir |= (byte)(1 << chipSelect);
+                if (value == PinValue.High)
+                {
+                    _gpioLowData |= (byte)(1 << chipSelect);
+                }
+                else
+                {
+                    byte mask = 0xFF;
+                    mask &= (byte)(~(1 << chipSelect));
+                    _gpioLowData &= mask;
+                }
+
+                for (int i = 0; i < NumberCycles; i++)
+                {
+                    toSend[idx++] = (byte)FtOpcode.SetDataBitsLowByte;
+                    toSend[idx++] = _gpioLowData;
+                    toSend[idx++] = _gpioLowDir;
+                }
+            }
+            else
+            {
+                _gpioHighDir |= (byte)(1 << (chipSelect - 8));
+                if (value == PinValue.High)
+                {
+                    _gpioHighData |= (byte)(1 << (chipSelect - 8));
+                }
+                else
+                {
+                    byte mask = 0xFF;
+                    mask &= (byte)(~(1 << (chipSelect - 8)));
+                    _gpioHighData &= mask;
+                }
+
+                for (int i = 0; i < NumberCycles; i++)
+                {
+                    toSend[idx++] = (byte)FtOpcode.SetDataBitsHighByte;
+                    toSend[idx++] = _gpioHighData;
+                    toSend[idx++] = _gpioHighDir;
+                }
+            }
+
+            Write(toSend);
+        }
+
+        #endregion
+
+        #region Read Write
+
+        internal void Write(ReadOnlySpan<byte> buffer)
+        {
+            uint numBytesWritten = 0;
+            var ftStatus = FtFunction.FT_Write(_ftHandle, in MemoryMarshal.GetReference(buffer), (ushort)buffer.Length, ref numBytesWritten);
+            if ((ftStatus != FtStatus.Ok) || (buffer.Length != numBytesWritten))
+            {
+                throw new IOException($"Can't write to the device");
+            }
+        }
+
+        internal int Read(Span<byte> buffer)
+        {
+            CancellationToken token = new CancellationTokenSource(1000).Token;
+            int totalBytesRead = 0;
+            uint bytesToRead = 0;
+            uint numBytesRead = 0;
+            FtStatus ftStatus;
+            while ((totalBytesRead < buffer.Length) && (!token.IsCancellationRequested))
+            {
+                bytesToRead = GetAvailableBytes();
+                if (bytesToRead > 0)
+                {
+                    ftStatus = FtFunction.FT_Read(_ftHandle, in buffer[totalBytesRead], bytesToRead, ref numBytesRead);
+                    if ((ftStatus != FtStatus.Ok) && (bytesToRead != numBytesRead))
+                    {
+                        throw new IOException("Can't read device");
+                    }
+
+                    totalBytesRead += (int)bytesToRead;
+                }
+            }
+
+            return totalBytesRead;
+        }
+
+        private bool FlushBuffer()
```

Comment:
since this means read all available bytes I'd suggest to call it: `DiscardInput()`